workflow-ui/compose/src/main/java/com/squareup/workflow1/ui/compose/LocalWorkflowEnvironment.kt

```diff
@@ -0,0 +1,10 @@
+package com.squareup.workflow1.ui.compose
+
+import androidx.compose.runtime.ProvidableCompositionLocal
+import androidx.compose.runtime.compositionLocalOf
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+
+@WorkflowUiExperimentalApi
+public val LocalWorkflowEnvironment: ProvidableCompositionLocal<ViewEnvironment> =
+  compositionLocalOf { ViewEnvironment.EMPTY }
```

Comment:
@zach-klippenstein @vgonda @ekeitho I was torn here between `compositionLocalOf` or `staticCompositionLocalOf`. Seemed like the former gave us a better hope of having bookkeeping in place that would reduce the impact of updates to the environment?

Reply:
yeah `staticCompositionLocalOf` should be used for things that are unlikely to change often, because it recomposes the entire scope that it's provided in. so things like `LocalDensity` is a good example usage of `staticCompositionLocalOf` - however `compositionLocalOf` would be a good use if it's value changes more frequently and so Compose does more tracking of where the value of the composition local is read. so if `ViewEnviornment` is changed frequently i'd say `compositionLocalOf` is the better option here.